File: src/main/java/kitchenpos/eatinorders/domain/tobe/OrderTableName.java
package kitchenpos.eatinorders.domain.tobe;

import javax.persistence.Embeddable;
import java.util.Objects;

@Embeddable
public class OrderTableName {

    private String name;

    protected OrderTableName() {
    }

    public OrderTableName(String name) {
        validateName(name);
        this.name = name;
    }

    private void validateName(String name) {
        if (Objects.isNull(name) || name.isEmpty()) {
            throw new IllegalArgumentException();
        }
    }

    public String getName() {
        return name;
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/domain/OrderTableName.kt
package kitchenpos.eatinorders.tobe.domain

@JvmInline
value class OrderTableName(
    val value: String
) {
    init {
        if (value.isEmpty()){
            throw IllegalArgumentException("주문 테이블 이름은 공백일 수 없습니다")
        }
    }
}
